lib/ukschedcoop/schedcoop.c

```diff
@@ -168,8 +171,11 @@ static void schedcoop_thread_woken(struct uk_sched *s, struct uk_thread *t)
 
 	if (t->wakeup_time > 0)
 		UK_TAILQ_REMOVE(&c->sleep_queue, t, queue);
-	if (t != uk_thread_current() && uk_thread_is_runnable(t))
+	if ((t != uk_thread_current() || uk_thread_is_queueable(t))
```

Comment:
Under the assumption that the comments are above are really issues:
I wonder if there is a case when `uk_thread_is_queuable(t)` does not imply `t != uk_thread_current`. (Except when the thread is initially created). Otherwise, the condition could be simply `uk_thread_is_queueable(t) && uk_thread_is_runnable(t)`